pkg/resource/bucket/hook.go

```diff
@@ -62,6 +63,11 @@ func (rm *resourceManager) createPutFields(
 			return err
 		}
 	}
+	if r.ko.Spec.Policy != nil {
+		if err := rm.syncPolicy(ctx, r); err != nil {
+			return err
+		}
+	}
```

Comment:
What if the `spec.policy` was deleted from the CR? `rm.deletePolicy(ctx, r)` will never be called i assume?

Reply:
This is in the `createPutFields` method, which is called after `CreateBucket`. For updates, we invoke `syncPolicy` through `customUpdateBucket`, which checks for `delta.DifferentAt("Spec.Policy")`.